### ISSUE TITLE
tests results html content per test tag in testng xml not as expected after upgrading to testng 7.1.0

### ISSUE THREAD

**Issue by hnestorova** [cbeust/testng#2364]
### TestNG Version
7.1.0

### Expected behavior
Each tests results html per test tag in testng xl contains only the results from the methods of the classes in the corresponding tag

### Actual behavior
If we have two test tags, the html tests results report for the second tag contains the result for the test in the class in the first tag as well 

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [X] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
To reproduce, create two classes with the below content and run the testng xml as suite for them
TestNG xml content:

```xml
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="TestClasses" parallel="classes" thread-count="1">
	<test verbose="1" name="Test1" annotations="JDK">
			<classes>
			<class name="com.some.testpackage.TestClass1"/>
			</classes>
	</test>
	<test verbose="1" name="Test2" annotations="JDK">
			<classes>
			<class name="com.some.testpackage.TestClass2"/>
			</classes>
	</test>
</suite>
```

![testNG](https://user-images.githubusercontent.com/30794708/92503299-bb58ae00-f209-11ea-8e19-4a772ad66ac6.PNG)

Classes content:

```java
package com.some.testpackage;

import org.testng.Assert;
import org.testng.annotations.Test;


public class TestClass1 {
	
	@Test
	public void case1() {
		Assert.assertTrue("".equals("a"));
	}
	
}
```

```java
package com.some.testpackage;

import org.testng.Assert;
import org.testng.annotations.Test;


public class TestClass2 {
	
	@Test
	public void case1() {
		Assert.assertTrue("".equals("a"));
	}	
}
```

**Comment by krmahadevan** [comment]
@hnestorova - The latest released version of TestNG is `7.3.0`. Please retry using that and post back your results

**Comment by hnestorova** [comment]
@krmahadevan indeed works as expected with the newer version, thank you!
